### PR TITLE
Add inactive timesteps option to Newmark beta so that time derivative calcul…

### LINKED ISSUE #18233: Newmark Beta time integrator gives large oscillations when run with a static initial step
## Bug Description
When applying gravity in structural dynamics problems, we need to disable the inertia kernels, and velocity and acceleration calculations in the first step, which involves performing a static calculation of Ku = F. When using the Newmark-Beta time integrator (which we most often use for dynamics) we cannot switch off time derivative (velocity and acceleration) calculations through the control system. Therefore, the time integrator will compute velocity and acceleration for the static step, which results in large spikes since the gravity step essentially leads to a sudden change in displacement. This combined with the properties of Newmark-Beta time integration results in noise in the whole simulation. 

## Steps to Reproduce
Perform a structural dynamics analysis with Newmark Beta TI with a static first step for gravity. Accelerations and velocities for this simulation will have large oscillations. 

## Impact
Currently, we cannot perform gravity analysis with a static step when we use the Newmark-Beta TI. If we do, the results will have large oscillations. The only other option is to slowly ramp gravity, which is very cumbersome and takes a long time. 

## Proposed solution
Add a parameter called `active_time` to the Newmark-Beta time integrator. If _t < _active_time, the derivates are set to zero. This way, we can avoid calculating the derivatives when there is a static step in the beginning. Rest is unchanged.

### PR DESCRIPTION
**Pull request by cbolisetti**
…ation can be avoided in the first few times steps, if needed. Closes #18233.

Following is an example with and without this PR. The orange plot corresponds to vertical accelerations in a model with just gravity, when accelerations are computed during the static step. The blue plot is the same acceleration when the acceleration computation is suppressed in the first time step using the `inactive_tsteps` parameter in NewmarkBeta.C. 

newmark_beta_noise

### CODE REVIEW

**Review by WilkAndy — COMMENTED** [review]
This needs to be documented, otherwise people won't understand why on earth they'd want `active_time`.   Basically just copy-and-paste your issue description into the markdown doco, along with that nice graph

Perhaps you should also say "this kernel returns zero" instead of "the time derivatives will be zero"

Also, use numeric_limits lowest instead of -1E100

**Comment by cbolisetti** [comment]
> This needs to be documented, otherwise people won't understand why on earth they'd want `active_time`. Basically just copy-and-paste your issue description into the markdown doco, along with that nice graph
> 
> Perhaps you should also say "this kernel returns zero" instead of "the time derivatives will be zero"
> 
> Also, use numeric_limits lowest instead of -1E100

Will do! Thank you, Andy.

**Comment by dschwen** [comment]
Shouldn't this parameter be in terms of steps rather than time? It seems to me you want to avoid taking the derivative in the first time step rather than within a certain time window.

**Comment by hugary1995** [comment]
Can you check in at least one regression test that fails without this option?

**Comment by hugary1995** [comment]
I'm not fully convinced that we actually need to do this to fix the issue in your problem -- but it's nice to have this feature regardless.

**Comment by cbolisetti** [comment]
Thanks @hugary1995 and @dschwen I will try to address your comments soon. 

@hugary1995 just curious, what alternative do you suggest to solve the problem without doing this?

**Comment by hugary1995** [comment]
My first thought is that the static step and the dynamic calculations should be separated. We should do a static step first with a steady executioner. Then it can be restarted using a transient executioner. I think this should work out-of-the-box within the moose framework, right?

Even if what I said above is correct, I have no objections in adding this `active_time` or maybe `active_time_step` parameter if that brings you convenience in practice.

**Comment by hugary1995** [comment]
Also, if you add this option, I think it better be added to all time integrators. Or even better, add it to the transient executioner. There's nothing special about Newmark-beta here.

**Comment by cbolisetti** [comment]
> Also, if you add this option, I think it'd better be added to all time integrators. Or even better, add it to the transient executioner. There's nothing special about Newmark-beta here.

Got it. A more permanent option would be to add the ability to switch time integrators, which, I was told is something they would like, but it will take some major changes.  Newmark-beta is a little bit special here because when you have a displacement spike in the first time step, the velocity and acceleration go haywire as in the figure above. Not sure if that is true for all time integrators. I'm adding this as a temporary thing, but a more permanent solution for all time integrators would be nice, I agree.

**Comment by cbolisetti** [comment]
@WilkAndy and @hugary1995, I think this PR is ready for you to take another look. Please let me know if this is ok. I am  relying on this being merged for another PR in MASTODON with a deadline, so if you can take a look at this sometime soon, I would really appreciate it.

**Comment by hugary1995** [comment]
Bottom line up front: I'm not on CCB...

Your code looks fine.

I'm still curious to know whether restart can solve your issue -- if it can, this PR will no longer hold you back, especially if you have a deadline.
If there's something that doesn't work with restart, then that's a separate issue we need to fix.

**Comment by cbolisetti** [comment]
@hugary1995 yes, we can use restart analysis, but currently its more cumbersome than running a static analysis in the first timestep. It is much easier to initialize the variables and auxvariables that we need if everything is in the same simulation and this PR enables us to do that.

File: framework/src/timeintegrators/NewmarkBeta.C [hunk 1]
```diff
@@ -23,13 +23,17 @@ NewmarkBeta::validParams()
...
+                       0,
+                       "The time derivatives will set to be zero for this number of time steps.");
   return params;
 }
 
 NewmarkBeta::NewmarkBeta(const InputParameters & parameters)
   : TimeIntegrator(parameters),
     _beta(getParam<Real>("beta")),
     _gamma(getParam<Real>("gamma")),
+    _inactive_tsteps(getParam<int>("inactive_tsteps")),
```
Comment (dschwen) [inline framework/src/timeintegrators/NewmarkBeta.C:36]: this should be unsigned

**Review by dschwen — APPROVED** [review]

File: framework/src/timeintegrators/NewmarkBeta.C (hunk 1 above)
Comment (dschwen) [reply framework/src/timeintegrators/NewmarkBeta.C:36]: `timeStep()` returns a signed integer, so nevermind

**Pull request merged**